Fix the JDBC timeout error in release mode

shalinilohia50 opened the pull request:
We were assigning a non-volatile variable "old_db_name" inside TRY earlier. Now we have moved assignment out of TRY block. 
In release mode, it was leading to segmentation fault in the case of runtime error earlier.

Task: BABEL-3077
Signed-off-by: Shalini Lohia <lshalini@amazon.com>

lohia-shalini committed:
Fix the JDBC timeout error in release mode

Moving the variable old_db_name out of try block. It was leading to
segmentation fault in the release mode.

Task: BABEL-3077
Signed-off-by: Shalini Lohia <lshalini@amazon.com>

surendravishnoi reviewed (changes requested):
Please add more details to commit message.

shalinilohia50 commented:
> 

Updated the commit message.

shalinilohia50 committed:
Empty commit to update the commit message

Moving the variable old_db_name out of try block.
In debug mode, "old_db_name" inside try block was not getting free
and hence at the runtime error we had the correct old_db_name.

But in release mode, "old_db_name" inside try block was getting free
and it was leading to segmentation fault in the case of runtime error.

Task: BABEL-3077
Signed-off-by: Shalini Lohia <lshalini@amazon.com>

shalinilohia50 committed:
We were assigning a non-volatile variable "old_db_name" inside TRY earlier. Now we have moved assignment out of TRY block.
